### ISSUE TITLE
Dasharo Openness Score improvements

### ISSUE THREAD

**Issue by pietrushnic** [Dasharo/dasharo-issues#276]
**The problem you're addressing (if any)**

Dasharo Openness Score is niche almost no one knows about it, despite the fact people asking questions about it.

**Describe the solution you'd like**

We should improve visual representation of Dasharo Openness Score as well as consider including  it as mandatory part of Dasharo Certification Program (together with other validation results e.g. .

**Where is the value to a user, and who might that user be?**

The value to users and whole  open-source firmware community is wide adoption and user friendly look and feel for standardize measure of openness. We often are asked about explaining level of openness with clear score we hope to delegate to nice looking charts and badges.

**Describe alternatives you've considered**

N/A

**Additional context**

**Comment by miczyg1** [comment]
Pasting a comment from Openness Score PR, because it is relevant:

- Script should run as part of CI/CD
- Dasharo Openness Score should be published as part of the newsletter.
- Dasharo Openness Score should be published as part of the DUG.
- We should target the production version for DUG#2.
- CI/CD should compare the Dasharo binary with reference binaries we obtained from hardware. We should not care about the privacy of hardware in the lab since it can already be considered public domain.
- Dasharo documentation should be adequately improved to explain our methodology.
- Versioning should be added. (milestones created, tags will be placed accordingly)

> For which releases and which platform do we plan to provide Openness Score?

For every possible platform that makes sense

**Comment by pietrushnic** [comment]
@macpijan @miczyg1 am I correct we will have CI/CD integration part as part of Novacustom integration?

**Comment by macpijan** [comment]
>  am I correct we will have CI/CD integration part as part of Novacustom integration?

Not really. We have integrated that in our release scripts, though.

**Comment by miczyg1** [comment]
Yes, so basically, we just addressed `Dasharo Openness Score should be published as part of the newsletter.` only for now

**Comment by miczyg1** [comment]
> I guess we have to clean that up. We did all that stuff without bumping the version of the Dasharo Openness Score, which is incorrect.

But we haven't changed a single bit in Openness Score code yet since v0.1... So it is no eligible for v0.2, neither the items in v0.2 milestone have been fulfilled

Quick status:

- Script should run as part of CI/CD - STAGING
- Dasharo Openness Score should be published as part of the newsletter. - DONE
- Dasharo Openness Score should be published as part of the DUG. - STAGING?
- We should target the production version for DUG#2. - We are past DUG#2, STAGING
- CI/CD should compare the Dasharo binary with reference binaries we obtained from hardware. We should not care about the privacy of hardware in the lab since it can already be considered public domain. - STAGING
- Dasharo documentation should be adequately improved to explain our methodology. - DONE already in v0.1: https://github.com/Dasharo/Openness-Score/blob/main/docs/source/methodology.rst
- Versioning should be added. (milestones created, tags will be placed accordingly) - DONE https://github.com/Dasharo/dasharo-issues/milestone/10, tag v0.1 was placed on 31st May when the code and documentation has been completed.

**Comment by pietrushnic** [comment]
> But we haven't changed a single bit in Openness Score code yet since v0.1... So it is no eligible for v0.2, neither the items in v0.2 milestone have been fulfilled

Because some requests are not exactly Openess Score code changes.

> Dasharo Openness Score should be published as part of the DUG. - STAGING?

During DUG#3 there will be the status presentation and DUG#4 should contain some reports for Dasharo releases.

> We should target the production version for DUG#2. - We are past DUG#2, STAGING

Integration with NovaCustom will be the first production run I guess MSI can be another.

**Comment by wessel-novacustom** [comment]
I think this has been integrated for NovaCustom laptops. I don't know for other devices that use Dasharo, I suggest to close the issue @pietrushnic if this is the case for all devices.

**Comment by pietrushnic** [comment]
@wessel-novacustom, thanks for the ping. @miczyg1 do you think we have to keep this issue? TBH, I'm not sure if anything from v0.2 was implemented.

**Comment by miczyg1** [comment]
Not all points have been addressed yet: https://github.com/Dasharo/dasharo-issues/issues/276#issuecomment-1738705507

**Comment by miczyg1** [comment]
@pietrushnic this point needs more clarification:

> CI/CD should compare the Dasharo binary with reference binaries we obtained from hardware. We should not care about the privacy of hardware in the lab since it can already be considered public domain

There are two main questions:
1. What information from two reports should we compare?
2. What form of comparison representation are we interested in?

It is not clear to me what we want to achieve here. Seems to be somewhat related to second bullet from this issue: https://github.com/Dasharo/dasharo-issues/issues/924 I gave some rough ideas on comparisons there. Let me know if there should be more to it.

**Comment by pietrushnic** [comment]
> What information from two reports should we compare?

I tried to answer that already here, confirming your POV:
https://github.com/Dasharo/dasharo-issues/issues/924#issuecomment-2548164882

From my perspective, the whole point is discussed in #924.

> What form of comparison representation are we interested in?

Does this also belong to #924? As I said in the linked comment, I don't know how. I would brainstorm with ChatGPT and create 2-3 versions of that comparison, show that to a couple of people, maybe someone from the community, and ask for feedback. It is tough to imagine what would be effective at this point. Maybe a stacked bar chart. The pie chart may be the same as before, but it is shown differently (side by side), so when you look at it, you can visually compare what changed in what area. A pie chart with some dotted lines shows how things looked before, and solid lines show the current state.

**Comment by DaniilKl** [comment]
> Script should run as part of CI/CD

@miczyg1, please confirm, that I understood the goal of this point correctly. The goal is to add a CI (in this or another repo) that will test the openness score tools for issues and regressions. This conclusion is based on [the PR the idea was initially publised](https://github.com/Dasharo/Openness-Score/pull/1). The sentence was actually written by @pietrushnic ([here](https://github.com/Dasharo/Openness-Score/pull/1#issuecomment-1484847018)), but you [stated](https://github.com/Dasharo/Openness-Score/pull/1#issuecomment-1485047194)  `Some (if not most) of these requests are already included in the first comment in the PR.`. Just to make sure I have understood the entire context here correctly.
